File: android/app/src/main/java/com/azad/sayit/OneSignalPlugin.java
package com.azad.sayit;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.onesignal.OneSignal;

@CapacitorPlugin(name = "OneSignalPlugin")
public class OneSignalPlugin extends Plugin {

    @PluginMethod
    public void login(PluginCall call) {
        String userId = call.getString("userId");
        if (userId == null || userId.isEmpty()) {
            call.reject("userId is required");
            return;
        }
        OneSignal.login(userId);
        call.resolve();
    }

    @PluginMethod
    public void logout(PluginCall call) {
        OneSignal.logout();
        call.resolve();
    }
}

File: ios/App/App/OneSignalPlugin.swift
import Capacitor
import OneSignalFramework

/// Thin Capacitor plugin that lets JavaScript link a user to OneSignal.
/// Named SayItOSBridge internally to avoid ObjC conflict with OneSignalFramework.
/// Call from JS: Capacitor.Plugins.SayItOSBridge.login({ userId: "<supabase-uuid>" })
@objc(SayItOSBridge)
public class SayItOSBridge: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "SayItOSBridge"
    public let jsName     = "SayItOSBridge"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "login",  returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "logout", returnType: CAPPluginReturnPromise),
    ]

    @objc func login(_ call: CAPPluginCall) {
        guard let userId = call.getString("userId"), !userId.isEmpty else {
            call.reject("userId is required")
            return
        }
        OneSignal.login(userId)
        print("[OneSignal] native login called for userId:", userId)
        call.resolve()
    }

    @objc func logout(_ call: CAPPluginCall) {
        OneSignal.logout()
        call.resolve()
    }
}
